spec/notification.js

```diff
@@ -1,7 +1,10 @@
 var expect = require('chai').expect, NotifyClient = require('../client/notification.js').NotifyClient,
+    MockDate = require('mockdate'),
     nock = require('nock'),
     createGovukNotifyToken = require('../client/authentication.js');
 
+MockDate.set(1234567890000);
```

Comment:
does this need to be here in the outer scope? given how everything is inside a `describe` callback, I'd be surprised if this does anything at all (esp since you have it in the beforeEach/afterEach)

Reply:
Trying to only set and unset the date in a `before` or `beforeEach` hook doesn’t work because we also use a ‘real’ date outside of any of the test cases. ([Here](https://github.com/alphagov/notifications-node-client/blob/961ad7ed4b595c92e870b0ed7408fb0483bc1336/spec/notification.js#L38) when we create the authentication Nock).

The only way I’ve been able to prevent the random failures is by mocking the date in both the outer scope as well as the test cases. It's not ideal, but I couldn't find a nicer way round it.